org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -589,6 +590,16 @@ public boolean setAutoBuilding(boolean enable) throws CoreException {
 
 								}
 							}
+							if (entry.getEntryKind() == IClasspathEntry.CPE_LIBRARY) {
+								IPath path = entry.getPath();
+								IFile resource = ResourcesPlugin.getWorkspace().getRoot().getFile(path);
+								if (resource != null && resource.exists() && !resource.isDerived()) {
+									IPath location = resource.getLocation();
+									if (location != null && !isContainedIn(location, sources)) {
+										sources.add(location);
```

Comment:
problem is you're later appending /** to the jar extension, which seems to cause some issues on my mac